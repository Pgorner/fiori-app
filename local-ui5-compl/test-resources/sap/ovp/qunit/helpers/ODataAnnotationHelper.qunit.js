sap.ui.define([
    "sap/ovp/helpers/ODataAnnotationHelper"
], function(ODataAnnotationHelper) {
    "use strict";

    QUnit.test("ODataAnnotationHelper - getRecords, get the annotation records for an entity type", function (assert) {
        var sAnnotationPath = "com.sap.vocabularies.UI.v1.Identification#navigationIntentBased";
        var aRecords = [
            {
                "SemanticObject":{"String":"browse"},
                "Action":{"String":"books1"},
                "RecordType":sAnnotationPath
            }
        ];
        var oEntityType = {
            "name": "Books",
            "com.sap.vocabularies.UI.v1.Identification#navigationIntentBased": aRecords
        };
        var oModel = {};
        assert.ok(ODataAnnotationHelper.getRecords(oEntityType, sAnnotationPath, oModel) === aRecords, "annotation records are returned");
    });
});
