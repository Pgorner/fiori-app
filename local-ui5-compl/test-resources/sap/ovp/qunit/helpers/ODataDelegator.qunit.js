sap.ui.define([
    "sap/ovp/helpers/ODataDelegator",
    "sap/ovp/cards/MetadataAnalyser",
    "sap/ovp/helpers/V4/MetadataAnalyzer"
], function (
    ODataDelegator,
    V2MetadataAnalyser,
    V4MetadataAnalyser
) {
    "use strict";

    QUnit.test("ODataDelegator - getParametersByEntitySet, should call / delegate responsiblity to the appropriate handler based on model version", function (assert) {
        var oEntitySet = {
            "name": "Books",
            "entityType": "CatalogService.Books",
            "Org.OData.Capabilities.V1.SearchRestrictions": {"Searchable": {"Bool": "false"}}
        };
        var v2spy = sinon.stub(V2MetadataAnalyser, "getParametersByEntitySet").callsFake(function (){return true;});
        var v4spy = sinon.stub(V4MetadataAnalyser, "getParametersByEntitySet").callsFake(function (){return true;});

        //v2 model
        var oModel = {
            getODataVersion: function () {
                return "2.0";
            }
        };
        ODataDelegator.getParametersByEntitySet(oModel, oEntitySet);
        assert.ok(v2spy.called, "V2 metadata analyzer gets called");

        //V4 model
        oModel.getODataVersion = function () {
            return "4.0";
        }
        ODataDelegator.getParametersByEntitySet(oModel, oEntitySet);
        assert.ok(v4spy.called, "V4 metadata analyzer gets called");

    });

    QUnit.test("ODataDelegator - getPropertyFromEntityType, should call / delegate responsiblity to the appropriate handler based on model version", function (assert) {
        var sPropertyPath = "category";
        var oEntityType = {
            "name": "Books",
            "property": [
                {
                    "name": "amount",
                    "type": "Edm.Int32"
                },
                {
                    "name": "category",
                    "type": "Edm.String"
                }
            ]
        }
        var v2spy = sinon.stub(V2MetadataAnalyser, "getPropertyFromEntityType").callsFake(function (){return true;});
        var v4spy = sinon.stub(V4MetadataAnalyser, "getPropertyFromEntityType").callsFake(function (){return true;});

        //v2 model
        var oModel = {
            getODataVersion: function () {
                return "2.0";
            }
        };
        ODataDelegator.getPropertyFromEntityType(sPropertyPath, oEntityType, oModel);
        assert.ok(v2spy.called, "V2 metadata analyzer gets called");
        
        //V4 model
        oModel.getODataVersion = function () {
            return "4.0";
        }
        ODataDelegator.getPropertyFromEntityType(sPropertyPath, oEntityType, oModel);
        assert.ok(v4spy.called, "V4 metadata analyzer gets called");

    });
});
