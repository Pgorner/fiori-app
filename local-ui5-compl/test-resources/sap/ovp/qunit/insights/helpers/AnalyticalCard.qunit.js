sap.ui.define([
    "sap/ovp/insights/helpers/AnalyticalCard",
    "test-resources/sap/ovp/qunit/cards/utils",
    "test-resources/sap/ovp/mockservers",
    "sap/ovp/insights/helpers/Batch",
    "test-resources/sap/ovp/qunit/cards/charts/utils",
    "sap/ovp/cards/charts/VizAnnotationManager",
    "sap/ovp/helpers/ODataAnnotationHelper",
], function (
    AnalyticalCard,
    Utils,
    mockservers,
    BatchHelper,
    ChartUtils,
    OvpVizAnnotationManager,
    ODataAnnotationHelper,
) {
    "use strict";
    QUnit.module("sap.ovp.insights.helpers.AnalyticalCard", {
        beforeEach: function () {
            mockservers.loadMockServer(Utils.odataBaseUrl_salesShare, Utils.odataRootUrl_salesShare);
        },
        afterEach: function () {
            mockservers.close();
        },
    });

    QUnit.test("AnalyticalCardHelper - createChartContent - validate created chart content.", function (assert) {
        var cardTestData = {
            card: {
                id: "chart_28",
                model: "salesShare",
                template: "sap.ovp.cards.charts.analytical",
                settings: {
                    entitySet: "SalesShare",
                    selectionAnnotationPath: "com.sap.vocabularies.UI.v1.SelectionVariant#Eval_by_CtryCurr",
                    chartAnnotationPath: "com.sap.vocabularies.UI.v1.Chart#Eval_by_Currency_Test_Forecast",
                    presentationAnnotationPath: "com.sap.vocabularies.UI.v1.PresentationVariant#Eval_by_CtryCurr",
                    dataPointAnnotationPath: "com.sap.vocabularies.UI.v1.DataPoint#Eval_by_CtryCurr",
                    identificationAnnotationPath: "com.sap.vocabularies.UI.v1.Identification#Eval_by_CtryCurr",
                }
            },
            dataSource: {
                baseUrl: ChartUtils.odataBaseUrl,
                rootUri: ChartUtils.odataRootUrl,
                annoUri: ChartUtils.testBaseUrl + "data/salesshare/annotations.xml",

            }
        };
        var oModel = Utils.createCardModel(cardTestData);
        var fnDone = assert.async();
        Utils.createCardView(cardTestData, oModel).then(function (oView) {
            var oCardController = oView.getController();
            var oCardViz = oView.byId("analyticalChart");

            oCardController.vizFrame = oCardViz;
            var oHandler = oCardViz && oCardViz.getParent();
            var oController = {
                getView: function () {
                    return null;
                },
                getOwnerComponent: function () {
                    return null;
                },
            };
            OvpVizAnnotationManager.buildVizAttributes(oCardViz, oHandler, oController);

            oView.getController().oMainComponent = {
                getGlobalFilter: function () {
                    return null;
                }
            };
            oView.getController().oCardComponentData.mainComponent = oView.getController().oMainComponent;
            var oCardComponentData = oCardController.oCardComponentData;
            var oCardDefinition = {
                vizFrame: oCardController.vizFrame,
                entitySet: "SalesShareDonut",
                entityType: oCardController.getEntityType(),
                cardComponentName: "Analytical",
                cardComponentData: oCardComponentData,
                cardComponent: oCardController.oCardComponent,
                view: oView
            };

            var oSapCard = {
                configuration: {
                    parameters: {}
                }
            };
            var getRequestUrlAndEntityInfoStub = sinon.stub(BatchHelper, "_getRequestUrlAndEntityInfo").returns({
                sPath: "SalesShareColumnStacked?$orderby=Region%20asc",
                _entitySet: {
                    value: "SalesShareColumnStacked"
                },
                _urlSuffix: {
                    value: ""
                }
            });
            var oEntityModel = oView.getModel();
            var oSettings = oCardDefinition.cardComponentData.settings;

            var oChartAnnotation = ODataAnnotationHelper.getRecords(oCardDefinition.entityType, oSettings.chartAnnotationPath, oEntityModel);
            var sTitle = oChartAnnotation.Title && oChartAnnotation.Title.String;

            var oChartContent = AnalyticalCard.createChartContent(oCardDefinition, oSettings.chartAnnotationPath, oSapCard);

            assert.deepEqual(oChartContent.chartType, oCardViz.getVizType(), "checking chart-type value");
            assert.deepEqual(oChartContent.title.text, sTitle, "checking for chart title value");
            assert.ok(
                oChartContent.title.visible == true,
                "Check if title is visible"
            );
            assert.deepEqual(oChartContent.dimensions.length, 1, "chart is having dimension as per configuration");

            assert.deepEqual(oChartContent.measures.length, oCardViz.getDataset().getMeasures().length, "chart is having measures as per configuration");
            assert.deepEqual(oChartContent.feeds.length, oCardViz.getFeeds().length, "chart is having feeds as per configuration");

            assert.deepEqual(oChartContent.dimensions[0].label, oCardViz.getDataset().getDimensions()[0].getName(), "checking dimensions value according to the entityset");
            assert.ok(
                oChartContent.measures[2].label === oCardViz.getDataset().getMeasures()[2].getName(),
                "Check if measure is being considered"
            );
            assert.ok(
                oCardViz.getVizProperties()["legend"].itemMargin === 1.25,
                "legends are set with the correct item margin"
            );
            getRequestUrlAndEntityInfoStub.restore();

            fnDone();
        });
    })
});
