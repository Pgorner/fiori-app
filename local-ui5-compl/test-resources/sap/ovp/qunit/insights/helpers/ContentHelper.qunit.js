sap.ui.define([
    "sap/ovp/insights/helpers/ContentHelper"
], function(ContentHelper) {
    "use strict";

    QUnit.test("ContentHelper  - function: addBrackets", function (assert) {
        assert.ok(ContentHelper.addBrackets("pathName") === "{pathName}", "Bracket is added to a path.");
        assert.ok(ContentHelper.addBrackets("{pathName}") === "{pathName}", "Additional unwanted bracket is not added to a path.");
        assert.ok(ContentHelper.addBrackets("{= ${pathName}}") === "{= ${pathName}}", "Additional unwanted bracket is not added to an expression.");
    });

    QUnit.test("ContentHelper  - function: getHeaderStatus", function (assert) {
        var oBatch;
        assert.deepEqual(ContentHelper.getHeaderStatus(oBatch), { text: "{= ${/d/__count} === '0' ? '' : extension.formatters.formatHeaderCount( ${/d/__count}) }" }, "Header status with count when there is call without batch.");
        oBatch = {batch: true};
        assert.deepEqual(ContentHelper.getHeaderStatus(oBatch), { "text": "{= ${/content/d/__count} === '0' ? '' : extension.formatters.formatHeaderCount( ${/content/d/__count}) }" }, "Header status with count when there is call with batch.");
    });

    QUnit.test("ContentHelper  - function: getData", function (assert) {
        var oBatch;
        assert.deepEqual(ContentHelper.getData(oBatch), { path: "/d/results" }, "Data path when there is call without batch.");
        oBatch = {batch: true};
        assert.deepEqual(ContentHelper.getData(oBatch), { path: "/content/d/results" }, "Data path when there is call with batch.");
    });

    QUnit.test("ContentHelper  - function: isAnExpression", function (assert) {
        assert.equal(ContentHelper.isAnExpression("{= ${pathName}}"), true, "String is an expression.");
        assert.equal(ContentHelper.isAnExpression("{pathName}"), false, "Path String is not an expression.");
        assert.equal(ContentHelper.isAnExpression("pathName"), false, "String is not an expression.");
    });

    QUnit.test("ContentHelper  - function: buildValueStateExpression", function (assert) {
        var sStateType = "color";
        assert.equal(ContentHelper.buildValueStateExpression("{= ${pathName}}"), "{= ${pathName}}", "Returns expression directly when expression is already present.");
        assert.equal(ContentHelper.buildValueStateExpression("Good", sStateType), "{= extension.formatters.formatCriticality('Good', 'color')}", "Returns new expression with value.");
        assert.equal(ContentHelper.buildValueStateExpression("{pathName}", sStateType), "{= extension.formatters.formatCriticality(${pathName}, 'color')}", "Returns new expression with path variable.");
    });

    QUnit.test("ContentHelper  - function: getBindingPathOrValue", function (assert) {
        // check empty
        var sPropertyName, sExpectedExpression;
        var oProperties = {
                property1: "property1Value", // vaue
                property2: "{property2Path}", // single path binding
                property3: {
                    parts: [{ path : '{property3Path}'}]
                },
                property4: {
                    parts: [{ path : '{property4Path}'}],
                    type: {
                        oFormat: {
                            oFormatOptions: {
                                "percentSign": "%",
                                "pattern": "#,##0.###"
                            }
                        }
                    }
                },
                property5: {
                    parts: [{ path : '{property5Path}'},
                        {
                            "value": {
                                "dateFormat": {
                                    "relative": true,
                                    "relativeScale": "auto",
                                    "bUTC": false
                                },
                                "bUTC": false,
                                "functionName": "CardAnnotationhelper.formatDate"
                            },
                            "model": "ovpCardProperties",
                            "mode": "OneWay"
                        }],
                    formatter: function() {}
                },
                property6: {
                    parts: [{ path : '{property6Path1}'}, { path : '{property6Path2}'},
                        {
                            "value": {
                                "numberOfFractionalDigits": "0",
                                "scaleFactor": "1000",
                                "functionName": "CardAnnotationhelper.formatCurrency"
                            },
                            "model": "ovpCardProperties",
                            "mode": "OneWay"
                        }],
                    formatter: function() {}
                },
                property7: {
                    parts: [{ path : '{property7Path1}'}, { path : '{property7Path2}'},
                        {
                            "value": {
                                "numberOfFractionalDigits": 1,
                                "functionName": "CardAnnotationhelper.formatNumberCalculation"
                            },
                            "model": "ovpCardProperties",
                            "mode": "OneWay"
                        }],
                    formatter: function() {}
                }
            };
        var oControl = {
                getProperty: function (propertyName) {
                    return oProperties[propertyName];
                },
                getBindingInfo: function () {
                    return;
                },
                getBindingPath: function (propertyName) {
                    return oProperties[propertyName];
                }
            };
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), null, "Returned null when parameters are empty.");
        // check direct value
        sPropertyName = "property1";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), "property1Value", "Returned property value when binding is not present.");
        // check simple binding
        oControl.getBindingInfo = function (propertyName) {
            return oProperties[propertyName];
        };
        sPropertyName = "property2";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), "{property2Path}", "Returned property path when simple binding is present.");
        // check binding with parts
        sPropertyName = "property3";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), "{property3Path}", "Returned property path when single path is present but in parts.");

        // check binding with parts and type with formatter
        sPropertyName = "property4";
        sExpectedExpression = "{= extension.formatters.formatNumber({'percentSign':'%','pattern':'#,##0.###'},[0],${{property4Path}})}";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), sExpectedExpression, "Returned expression with formatting from 'type'.");

        // date formatter
        sPropertyName = "property5";
        sExpectedExpression = "{= extension.formatters.formatDate(${{property5Path}}, {'relative':true,'relativeScale':'auto','bUTC':false})}";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), sExpectedExpression, "Returned expression with date formatter");

        // currency formatter
        sPropertyName = "property6";
        sExpectedExpression = "{= extension.formatters.formatCurrency({'numberOfFractionalDigits':'0','scaleFactor':'1000'},false,${{property6Path1}},${{property6Path2}})}";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), sExpectedExpression, "Returned expression with currency formatter");

        // formatNumberCalculation formatter
        sPropertyName = "property7";
        sExpectedExpression = "{= extension.formatters.formatNumber({'numberOfFractionalDigits':1},[0],${{property7Path1}},${{property7Path2}})}";
        assert.equal(ContentHelper.getBindingPathOrValue(oControl, sPropertyName), sExpectedExpression, "Returned expression with formatNumberCalculation formatter");

    });  
});
