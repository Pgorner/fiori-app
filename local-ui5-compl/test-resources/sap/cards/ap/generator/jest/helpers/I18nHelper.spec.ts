/*!
 * SAP UI development toolkit for HTML5 (SAPUI5)
 *  * (c) Copyright 2009-2025 SAP SE. All rights reserved
 */
import { createAndStoreGeneratedi18nKeys, resolveI18nTextFromResourceModel } from "sap/cards/ap/generator/helpers/I18nHelper";
import { CardManifest } from "sap/ui/integration/widgets/Card";

describe("I18nHelper, resolveI18nTextFromResourceModel", () => {
	test("should resolve i18n text from resource model", () => {
		const key = "{{key}}";
		const resourceModel = {
			getObject: jest.fn().mockReturnValue("value")
		};
		const result = resolveI18nTextFromResourceModel(key, resourceModel);
		expect(result).toBe("value");
		expect(resourceModel.getObject).toHaveBeenCalledWith("key");
	});

	test("should return key if it does not start with '{{' and end with '}}'", () => {
		const key = "key";
		const resourceModel = {
			getObject: jest.fn().mockReturnValue("value")
		};
		const result = resolveI18nTextFromResourceModel(key, resourceModel);
		expect(result).toBe(key);
		expect(resourceModel.getObject).not.toHaveBeenCalled();
	});
});

describe("I18nHelper, createAndStoreGeneratedi18nKeys", () => {
	test("Manifest is generated with I18N properties, with bindings", () => {
		const cardManifest: CardManifest = {
			_version: "1.15.0",
			"sap.app": {
				id: "objectCard",
				type: "card",
				title: "{card_title}",
				subTitle: "{card_subtitle}",
				description: "{card_description}",
				applicationVersion: {
					version: "1.0.0"
				}
			},
			"sap.card": {
				type: "Object",
				configuration: {
					parameters: {
						footerActionParameters: {
							"com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)":
								{},
							"com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)":
								{}
						},
						_adaptiveFooterActionParameters: {
							"com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)":
								{
									style: "default",
									verb: "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock",
									label: "Set Billing Block",
									actionParameters: [
										{
											isRequired: true,
											id: "HeaderBillingBlockReason",
											label: "Billing Block",
											errorMessage: "This is a required input",
											placeholder: "Please choose",
											configuration: {
												entitySet: "I_BillingBlockReason",
												serviceUrl:
													"/sap/opu/odata4/sap/c_salesordermanage_srv/srvd_f4/sap/i_billingblockreason/0001;ps=%27srvd-c_salesordermanage_sd-0001%27;va=%27com.sap.gateway.srvd.c_salesordermanage_sd.v0001.ae-c_salesordermanage.setbillingblock.headerbillingblockreason.SalesOrderManageType%27/I_BillingBlockReason?$select=BillingBlockReason,BillingBlockReason_Text&skip=0&$top=20",
												value: "${BillingBlockReason_Text} (${BillingBlockReason})",
												title: "${BillingBlockReason}"
											}
										}
									],
									data: {
										isConfirmationRequired: false
									},
									enablePath: "__OperationControl/SetBillingBlock",
									complimentaryActionKey: "",
									isComplimentaryAction: false,
									triggerActionText: "OK"
								},
							"com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)":
								{
									style: "default",
									verb: "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock",
									label: "Set Delivery Block",
									actionParameters: [
										{
											isRequired: true,
											id: "DeliveryBlockReason",
											label: "Delivery Block",
											errorMessage: "This is a required input",
											placeholder: "Please choose",
											configuration: {
												entitySet: "I_DeliveryBlockReason",
												serviceUrl:
													"/sap/opu/odata4/sap/c_salesordermanage_srv/srvd_f4/sap/i_deliveryblockreason/0001;ps=%27srvd-c_salesordermanage_sd-0001%27;va=%27com.sap.gateway.srvd.c_salesordermanage_sd.v0001.ae-c_salesordermanage.setdeliveryblock.deliveryblockreason.SalesOrderManageType%27/I_DeliveryBlockReason?$select=DeliveryBlockReason,DeliveryBlockReason_Text&skip=0&$top=20",
												value: "${DeliveryBlockReason_Text} (${DeliveryBlockReason})",
												title: "${DeliveryBlockReason}"
											}
										}
									],
									data: {
										isConfirmationRequired: false
									},
									enablePath: "__OperationControl/SetDeliveryBlock",
									complimentaryActionKey: "",
									isComplimentaryAction: false,
									triggerActionText: "OK"
								}
						}
					}
				},
				header: {
					title: "{sales_order_id}",
					data: {}
				},
				data: {},
				content: {
					data: {
						path: ""
					},
					groups: [
						{
							title: "Group Title",
							items: [
								{
									label: "Sales Order ID",
									value: "{sales_order_id}"
								}
							]
						}
					]
				},
				footer: {
					actionsStrip: [
						{
							type: "Button",
							visible: false,
							text: "Set Billing Block",
							buttonType: "Default",
							actions: [
								{
									type: "custom",
									enabled: "${__OperationControl/SetBillingBlock}",
									parameters:
										"{{parameters.footerActionParameters.com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)}}"
								}
							]
						},
						{
							type: "Button",
							visible: false,
							text: "Set Delivery Block",
							buttonType: "Default",
							actions: [
								{
									type: "custom",
									enabled: "${__OperationControl/SetDeliveryBlock}",
									parameters:
										"{{parameters.footerActionParameters.com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)}}"
								}
							]
						}
					]
				}
			}
		};

		createAndStoreGeneratedi18nKeys(cardManifest);
		expect(cardManifest).toMatchSnapshot();
	});

	test("Manifest is generated with I18N properties, with values", () => {
		const cardManifest: CardManifest = {
			_version: "1.15.0",
			"sap.app": {
				id: "objectCard",
				type: "card",
				title: "Card Title",
				subTitle: "Card Subtitle",
				description: "Card Description",
				applicationVersion: {
					version: "1.0.0"
				}
			},
			"sap.card": {
				type: "Object",
				header: {
					title: "Sales Order ID",
					data: {}
				},
				data: {},
				content: {
					data: {
						path: ""
					},
					groups: [
						{
							title: "Group Title",
							items: [
								{
									label: "Sales Order ID",
									value: "{sales_order_id}"
								},
								{
									label: "Sales Order ID",
									value: "{sales_order_id_2}"
								}
							]
						},
						{
							title: "",
							items: [
								{
									label: "",
									value: "{sales_order_id}"
								}
							]
						}
					]
				}
			}
		};

		createAndStoreGeneratedi18nKeys(cardManifest);
		expect(cardManifest).toMatchSnapshot();
	});
});
