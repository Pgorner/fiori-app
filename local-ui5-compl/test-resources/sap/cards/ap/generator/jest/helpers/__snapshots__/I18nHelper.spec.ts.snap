// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nHelper, createAndStoreGeneratedi18nKeys Manifest is generated with I18N properties, with bindings 1`] = `
{
  "_version": "1.15.0",
  "sap.app": {
    "applicationVersion": {
      "version": "1.0.0",
    },
    "description": "{card_description}",
    "id": "objectCard",
    "subTitle": "{card_subtitle}",
    "title": "{card_title}",
    "type": "card",
  },
  "sap.card": {
    "configuration": {
      "parameters": {
        "_adaptiveFooterActionParameters": {
          "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)": {
            "actionParameters": [
              {
                "configuration": {
                  "entitySet": "I_BillingBlockReason",
                  "serviceUrl": "/sap/opu/odata4/sap/c_salesordermanage_srv/srvd_f4/sap/i_billingblockreason/0001;ps=%27srvd-c_salesordermanage_sd-0001%27;va=%27com.sap.gateway.srvd.c_salesordermanage_sd.v0001.ae-c_salesordermanage.setbillingblock.headerbillingblockreason.SalesOrderManageType%27/I_BillingBlockReason?$select=BillingBlockReason,BillingBlockReason_Text&skip=0&$top=20",
                  "title": "\${BillingBlockReason}",
                  "value": "\${BillingBlockReason_Text} (\${BillingBlockReason})",
                },
                "errorMessage": "{{CardGenerator_AdaptiveCardAction_ActionParameterErrorMsg}}",
                "id": "HeaderBillingBlockReason",
                "isRequired": true,
                "label": "{{CardGenerator_AdaptiveCardAction_0_ActionParameterLabel_0}}",
                "placeholder": "{{CardGenerator_AdaptiveCardAction_ActionParameterPlaceholder}}",
              },
            ],
            "complimentaryActionKey": "",
            "data": {
              "isConfirmationRequired": false,
            },
            "enablePath": "__OperationControl/SetBillingBlock",
            "isComplimentaryAction": false,
            "label": "{{CardGenerator_AdaptiveCardActions_0_Label}}",
            "style": "default",
            "triggerActionText": "{{CardGenerator_AdaptiveCardActions_OkButton}}",
            "verb": "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock",
          },
          "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)": {
            "actionParameters": [
              {
                "configuration": {
                  "entitySet": "I_DeliveryBlockReason",
                  "serviceUrl": "/sap/opu/odata4/sap/c_salesordermanage_srv/srvd_f4/sap/i_deliveryblockreason/0001;ps=%27srvd-c_salesordermanage_sd-0001%27;va=%27com.sap.gateway.srvd.c_salesordermanage_sd.v0001.ae-c_salesordermanage.setdeliveryblock.deliveryblockreason.SalesOrderManageType%27/I_DeliveryBlockReason?$select=DeliveryBlockReason,DeliveryBlockReason_Text&skip=0&$top=20",
                  "title": "\${DeliveryBlockReason}",
                  "value": "\${DeliveryBlockReason_Text} (\${DeliveryBlockReason})",
                },
                "errorMessage": "{{CardGenerator_AdaptiveCardAction_ActionParameterErrorMsg}}",
                "id": "DeliveryBlockReason",
                "isRequired": true,
                "label": "{{CardGenerator_AdaptiveCardAction_1_ActionParameterLabel_0}}",
                "placeholder": "{{CardGenerator_AdaptiveCardAction_ActionParameterPlaceholder}}",
              },
            ],
            "complimentaryActionKey": "",
            "data": {
              "isConfirmationRequired": false,
            },
            "enablePath": "__OperationControl/SetDeliveryBlock",
            "isComplimentaryAction": false,
            "label": "{{CardGenerator_AdaptiveCardActions_1_Label}}",
            "style": "default",
            "triggerActionText": "{{CardGenerator_AdaptiveCardActions_OkButton}}",
            "verb": "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock",
          },
        },
        "footerActionParameters": {
          "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)": {},
          "com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)": {},
        },
      },
    },
    "content": {
      "data": {
        "path": "",
      },
      "groups": [
        {
          "items": [
            {
              "label": "{{CardGeneratorGroupPropertyLabel_Groups_0_Items_0}}",
              "value": "{sales_order_id}",
            },
          ],
          "title": "{{CardGeneratorGroupHeader_Groups_0}}",
        },
      ],
    },
    "data": {},
    "footer": {
      "actionsStrip": [
        {
          "actions": [
            {
              "enabled": "\${__OperationControl/SetBillingBlock}",
              "parameters": "{{parameters.footerActionParameters.com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetBillingBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)}}",
              "type": "custom",
            },
          ],
          "buttonType": "Default",
          "text": "{{CardGenerator_AdaptiveCardActions_0_Label}}",
          "type": "Button",
          "visible": false,
        },
        {
          "actions": [
            {
              "enabled": "\${__OperationControl/SetDeliveryBlock}",
              "parameters": "{{parameters.footerActionParameters.com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SetDeliveryBlock(com.sap.gateway.srvd.c_salesordermanage_sd.v0001.SalesOrderManageType)}}",
              "type": "custom",
            },
          ],
          "buttonType": "Default",
          "text": "{{CardGenerator_AdaptiveCardActions_1_Label}}",
          "type": "Button",
          "visible": false,
        },
      ],
    },
    "header": {
      "data": {},
      "title": "{sales_order_id}",
    },
    "type": "Object",
  },
}
`;

exports[`I18nHelper, createAndStoreGeneratedi18nKeys Manifest is generated with I18N properties, with values 1`] = `
{
  "_version": "1.15.0",
  "sap.app": {
    "applicationVersion": {
      "version": "1.0.0",
    },
    "description": "Card Description",
    "id": "objectCard",
    "subTitle": "Card Subtitle",
    "title": "{{CardGeneratorHeaderTitle}}",
    "type": "card",
  },
  "sap.card": {
    "content": {
      "data": {
        "path": "",
      },
      "groups": [
        {
          "items": [
            {
              "label": "{{CardGeneratorGroupPropertyLabel_Groups_0_Items_0}}",
              "value": "{sales_order_id}",
            },
            {
              "label": "{{CardGeneratorGroupPropertyLabel_Groups_0_Items_1}}",
              "value": "{sales_order_id_2}",
            },
          ],
          "title": "{{CardGeneratorGroupHeader_Groups_0}}",
        },
        {
          "items": [
            {
              "label": "",
              "value": "{sales_order_id}",
            },
          ],
          "title": "",
        },
      ],
    },
    "data": {},
    "header": {
      "data": {},
      "title": "{{CardGeneratorHeaderTitle}}",
    },
    "type": "Object",
  },
}
`;
